### PR TITLE
csound: add Python bindings

### PR DESCRIPTION
**Pull request by nwhetsell**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This adds Python bindings for Csound.

This is another attempt at PR #39393.

### CODE REVIEW

**Comment by nwhetsell** [comment]
Anything I can do to help get this merged?

**Comment by zbeekman** [comment]
Sorry, this slipped through the cracks...

**Comment by zbeekman** [comment]
@BrewTestBot test this please

File: Formula/csound.rb [hunk 1]
```diff
@@ -29,18 +33,39 @@ def install
...
 
     mkdir "build" do
       system "cmake", "..", *args
       system "make", "install"
 
-      include.install_symlink "#{frameworks}/CsoundLib64.framework/Headers" => "csound"
+      macho = MachO.open("_csnd6.so")
+      macho.change_install_name("@rpath/CsoundLib64.framework/Versions/6.0/CsoundLib64", "#{frameworks}/CsoundLib64.framework/CsoundLib64")
+      macho.change_install_name("@rpath/libcsnd6.6.0.dylib", "#{lib}/libcsnd6.dylib")
+      macho.delete_rpath(pwd.sub(%r{^/private/}, "/"))
```
Comment (zbeekman) [inline Formula/csound.rb]: Shouldn't upstream be responsible for providing a sane build system with proper install names? I don't think it make sense for us to be doing this much tweaking of the installation.

Comment (nwhetsell) [reply Formula/csound.rb]: I’d say yes. If this is a deal-breaker here, I can open a pull request upstream.

Comment (zbeekman) [reply Formula/csound.rb]: Yup, I think this is a deal breaker for us. A pr upstream with a patch here (once they show interest in merging it) is far preferable.

Thanks!

Comment (nwhetsell) [reply Formula/csound.rb]: OK, thanks for the review.

Comment (nwhetsell) [reply Formula/csound.rb]: On second thought, one of the two Python bindings doesn’t require changing the installation like this, so this is now removed, but this appears to be one-step-forward-and-two-steps back—the test failure is because of the recently added audit for linking against Accelerate (https://github.com/Homebrew/brew/commit/35a271e99b656d49c0fd44356a68558143942aed).

**Comment by zbeekman** [comment]
Please add `depends_on “openblas”` and then add `-DBLA_VENDOR=OpenBLAS` to
the cmake flags.

**Comment by nwhetsell** [comment]
I may have done something wrong (the changes I made are at https://github.com/nwhetsell/homebrew-core/commit/100b58701608db3c40dbe4c27b8b52c4c364bfde), but I have a feeling this is another upstream problem. It looks like using Accelerate is somehow hard-coded—

https://github.com/csound/csound/blob/8797077654794c1ed78cac640f4485f49b77c217/CMakeLists.txt#L1308-L1315

—and searches of the upstream repository for [`DBLA_VENDOR`](https://github.com/csound/csound/search?q=DBLA_VENDOR) or [`BLA_VENDOR`](https://github.com/csound/csound/search?q=BLA_VENDOR) come up empty.

**Comment by zbeekman** [comment]
BLA_VENDOR controls the FindBLAS and FindLAPACK built in cmake modules.
This is how most people locate these libraries with CMake, and reduced
maintenance associated with finding appropriate linear algebra libs by
shifting that responsibility to CMake.

I wonder if upstream would consider using this standard mechanism rather
than finding accelerate.

If not, I think it may be worth creating an exception to this audit.

**Comment by zbeekman** [comment]
@bnwhetsell: Upon further investigation this appears to be a bug in Homebrew. Accelerate provides more than just BLAS/LAPACK, so this should not create an error. We need alternate logic to handle this correctly, probably a combination of backing in `-DBLA_VENDOR` to `std_cmake_args` and auditing for VecLibFort.

**Comment by nwhetsell** [comment]
OK, thanks. Should I revert the OpenBLAS-related change then?

**Comment by zbeekman** [comment]
Yes please. I'll work on patching homebrew/brew to behave in a saner manner. The audit failure will be unavoidable, but we may merge anyway, or run it through again after the fix.

File: Formula/csound.rb [hunk 2]
```diff
@@ -68,5 +88,10 @@ def install
     assert_equal "hello, world\n", stdout
     assert_match /^rtaudio:/, stderr
     assert_match /^rtmidi:/, stderr
+
+    ENV["DYLD_FRAMEWORK_PATH"] = "#{opt_prefix}/Frameworks"
```
Comment (zbeekman) [inline Formula/csound.rb]: Why is this needed?
Doesn't a symbolic link get installed into /usr/local/Frameworks?

Comment (nwhetsell) [reply Formula/csound.rb]: This is needed because otherwise Python can’t find CsoundLib64.framework.

As far as I know, how dynamically-linked frameworks are found is described in the documentation of `dyld`. From `man dyld`:

> For each framework that a program uses, the dynamic linker looks for the framework in each directory in DYLD_FRAMEWORK_PATH in turn. If it looks in all the directories and can’t find the framework, it searches the directories in DYLD_LIBRARY_PATH in turn. If it still can’t find the framework, it then searches DYLD_FALLBACK_FRAMEWORK_PATH and DYLD_FALLBACK_LIBRARY_PATH in turn.

I believe DYLD_FRAMEWORK_PATH and DYLD_LIBRARY_PATH are both undefined (or empty) by default. The `dyld` man page says this about DYLD_FALLBACK_FRAMEWORK_PATH:

> **DYLD_FALLBACK_FRAMEWORK_PATH**
>
> This is a colon separated list of directories that contain frameworks. It is used as the default location for frameworks not found in their install path. By default, it is set to /Library/Frameworks:/Network/Library/Frameworks:/System/Library/Frameworks

The `dyld` man page says this about DYLD_FALLBACK_LIBRARY_PATH:

> **DYLD_FALLBACK_LIBRARY_PATH**
>
> This is a colon separated list of directories that contain libraries. It is used as the default location for libraries not found in their install path. By default, it is set to $(HOME)/lib:/usr/local/lib:/lib:/usr/lib.

Consequently, by default frameworks are searched for in these folders:

1. /Library/Frameworks
2. /Network/Library/Frameworks
3. /System/Library/Frameworks
4. $(HOME)/lib
5. /usr/local/lib
6. /lib
7. /usr/lib

CsoundLib64.framework does get installed in /usr/local/Frameworks, but that’s not one of the default search paths.

**Comment by nwhetsell** [comment]
I’ve now reverted the OpenBLAS change. Thanks again for your review!

**Comment by zbeekman** [comment]
@nwhetsell is this ready to ship? I still need to PR against homebrew/brew for the audit failure, but let me know when this is ready, please.

**Comment by nwhetsell** [comment]
Yes, this is ready, sorry about the numpy oversight.

**Comment by zbeekman** [comment]
no problem

**Comment by zbeekman** [comment]
Sorry, I was trying to rebase your branch onto current homebrew/homebrew-core:master but messed up.

For future reference, making changes on `master` even on your own fork is an anti-pattern. (To be clear, the mess up was in fact my fault, but it would have been less complicated for me if you were working on a different branch.) I'm opening a PR with your changes from my fork (after squashing).

**Comment by nwhetsell** [comment]
OK, sorry about that.

**Comment by zbeekman** [comment]
No problem. My fault too.
